Add CEReactions tests

xiuqijix opened the pull request:
part of issue #5727
- HTMLMapElement
- HTMLCanvasElement
- HTMLDataElement
- HTMLDetailsElement
- HTMLModElement
- HTMLOListElement
- HTMLQuoteElement
- HTMLSlotElement
- HTMLTimeElement

xiuqijix committed:
Add CEReactions tests

- HTMLMapElement
- HTMLCanvasElement
- HTMLDataElement
- HTMLDetailsElement
- HTMLModElement
- HTMLOListElement
- HTMLQuoteElement
- HTMLSlotElement
- HTMLTimeElement

domenic commented:
w3c-test:mirror

domenic commented on custom-elements/reactions/HTMLDetailsElement.html:
```diff
@@ -0,0 +1,15 @@
+
+<script>
+
+testReflectAttribute('open', 'open', 'true', 'false', 'open on HTMLDetailsElement', 'details', HTMLDetailsElement);
```
Boolean attributes don't get reflected by the strings "true" and "false". They get reflected by the removal or addition of the attribute.

domenic commented on custom-elements/reactions/HTMLModElement.html:
```diff
@@ -0,0 +1,18 @@
+
+<script>
+
+testReflectAttribute('cite', 'cite', '../resources/custom-elements-helpers.js', './resources/reactions.js', 'cite on ins use HTMLModElement', 'ins', HTMLModElement);
```
This test fails on Chrome. I believe it is because the attribute name is `datetime`, not `dateTime`.

domenic commented on custom-elements/reactions/HTMLOListElement.html:
```diff
@@ -0,0 +1,17 @@
+
+<script>
+
+testReflectAttribute('reversed', 'reversed', 'true', 'false', 'reversed on HTMLOListElement', 'ol', HTMLOListElement);
```
Boolean attributes problem again

domenic commented on custom-elements/reactions/HTMLTimeElement.html:
```diff
@@ -0,0 +1,15 @@
+
+<script>
+
+testReflectAttribute('dateTime', 'dateTime', '2018-12-10', '2018-12-12', 'dateTime on HTMLTimeElement', 'time', HTMLTimeElement);
```
Also fails on Chrome, also I guess for the same reason.

domenic reviewed (changes requested):
Tests are wrong with boolean attributes; unclear what's going on with

xiuqijix committed:
Address @domenic's comments

- Use "datetime" as the attribute name.
- Modify the tests for boolean attributes.

kaixinjxq commented:
@domenic, many thanks for your review! I just submit a new commit to address your comments, PTAL.

domenic commented:
w3c-test:mirror
